third_party/scsi_mock/scsi_mock_module.c

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 Google LLC
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// version 2 as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+#include "scsi_mock_module.h"
+
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/scatterlist.h>
+#include <scsi/scsi.h>
+#include <scsi/scsi_host.h>
+
+static int scsi_queuecommand(struct Scsi_Host* host, struct scsi_cmnd* cmd) {
+  printk("RECIEVED COMMAND");
+  return 0;
+}
+
+static int scsi_abort(struct scsi_cmnd* cmd) { return SUCCESS; }
+
+static const char* scsi_mock_info(struct Scsi_Host* host) {
+  return "SCSI Mock Host, Version 0.1";
```

Comment:
Consider defining a macro for the module version so that you can use it here (via raw string concatenation) and in the registration macro at the end of the file.

Reply:
Good point, that's definitely more maintainable! I've gone ahead and added that feature